pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -927,3 +915,42 @@ ${compileResult.stdout}
 extension on Uri {
   Uri get parent => File(toFilePath()).parent.uri;
 }
+
+/// Parses depfile contents.
+///
+/// Format: `path/to/my.dill: path/to/my.dart, path/to/more.dart`
+///
+/// However, the spaces in paths are escaped with backslashes, and the
+/// backslashes are escaped with backslashes:
+///
+/// ```dart
+/// String _escapePath(String path) {
+///   return path.replaceAll('\\', '\\\\').replaceAll(' ', '\\ ');
+/// }
+/// ```
+List<String> parseDepFileInputs(String contents) {
+  final output = contents.split(': ').first;
+  contents = contents.substring(output.length + ': '.length).trim();
+  final inputs = <String>[];
+  var currentWord = '';
+  var escape = false;
+  for (var i = 0; i < contents.length; i++) {
+    if (contents[i] == r'\' && !escape) {
+      escape = true;
+    } else if (contents[i] == ' ' && !escape) {
+      inputs.add(currentWord);
+      currentWord = '';
+    } else {
+      currentWord += contents[i];
```

Comment:
This uses string concatentation, one character at a time. This is `O(n^2)`. Imagine a long path of 100 characters, this would be 5-10.000 operations.

Consider writing this something like this
```dart
while (int index < contents.length) {
  // Scan one part
  int start = index;
  while (index < contents.length) {
     final u = contents.codeUnitAt(index);
     if (u == ' '.codeUnitAt(0)) break;

     if (u == '\\'.codeUnitAt(0)) {
       index++; // skip backslash
       if (index == contents.length) throw 'malformed, ending with backslash';
       index++; // include escaped space. or backslash (could check for them here?)
       continue;
     }
     
    // normal character that's part of the string, include it.
    index++;
  }

  // Add the scanned part 
  parts.add(contents.substring(start, index));
}
```